src/SFA.DAS.ApprenticeFeedback.Application/Queries/GetApprenticeFeedbackTargetsForUpdate/GetApprenticeFeedbackTargetsForUpdateQueryHandler.cs

```diff
@@ -0,0 +1,70 @@
+﻿using MediatR;
+using SFA.DAS.ApprenticeFeedback.Domain.Configuration;
+using SFA.DAS.ApprenticeFeedback.Domain.Interfaces;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using static SFA.DAS.ApprenticeFeedback.Domain.Models.Enums;
+
+namespace SFA.DAS.ApprenticeFeedback.Application.Queries.GetApprenticeFeedbackTargetsForUpdate
+{
+    public static class ApprenticeFeedbackTargetFilters
+    {
+        public static IQueryable<Domain.Entities.ApprenticeFeedbackTarget> HasStarted(this IQueryable<Domain.Entities.ApprenticeFeedbackTarget> afts, IDateTimeHelper dateTimeHelper)
+        {
+            return afts.Where(aft => aft.StartDate.HasValue && aft.StartDate.Value.Date < dateTimeHelper.Now.Date);
+        }
+
+        public static IQueryable<Domain.Entities.ApprenticeFeedbackTarget> StatusNotCompletedOrPermanentlyWithdrawn(this IQueryable<Domain.Entities.ApprenticeFeedbackTarget> afts)
+        {
+            return afts.Where(aft => aft.Status != (int)FeedbackTargetStatus.Complete && aft.Status != (int)FeedbackTargetStatus.Withdrawn);
+        }
+        public static IQueryable<Domain.Entities.ApprenticeFeedbackTarget> FeedbackEligibilityNotCalulcatedRecently(this IQueryable<Domain.Entities.ApprenticeFeedbackTarget> afts, IDateTimeHelper dateTimeHelper, ApplicationSettings appSettings)
+        {
+            return afts.Where(aft => !(aft.EligibilityCalculationDate.HasValue && aft.EligibilityCalculationDate.Value.Date.AddDays(appSettings.EligibilityCalculationThrottleDays) > dateTimeHelper.Now.Date));
+        }
+        public static IQueryable<Domain.Entities.ApprenticeFeedbackTarget> NotGivenFeedbackRecently(this IQueryable<Domain.Entities.ApprenticeFeedbackTarget> afts, IDateTimeHelper dateTimeHelper, ApplicationSettings appSettings)
+        {
+            return afts.Where(aft => aft.ApprenticeFeedbackResults.OrderByDescending(a => a.DateTimeCompleted).FirstOrDefault() == null? 
+                false 
+                : !( aft.ApprenticeFeedbackResults.OrderByDescending(a => a.DateTimeCompleted).FirstOrDefault().DateTimeCompleted.Value.Date.AddDays(appSettings.EligibilityCalculationThrottleDays) > dateTimeHelper.Now.Date));
```

Comment:
I think you've got the wrong configuration in here, it's not the Eligibility Calculation Throttle, it's the RecentFeedback config.

In addition, I think it might be this one that may give you trouble in terms of, if you've tried a SQL profiler, is it leveraging the database, or pulling the entire data set back, in which case we might need to have a bit of a rethink.